tests/unit/rate_limiting/test_integration.py

```
@@ -0,0 +1,297 @@
+        
+        assert "gemini-2.5-flash" in str(exc_info.value)
+        assert exc_info.value.model == "gemini-2.5-flash"
+    
+    @pytest.mark.asyncio
+    async def test_circuit_breaker_activation(self, rate_limiter, mock_redis):
+        """Test circuit breaker activation after failures."""
+        async def failing_api_call():
+            raise Exception("API Error")
+        
+        # First failure should not open circuit
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)